### PR TITLE
Implement preferred currency, and appropriate conversion rate

### PR DESCRIPTION
**Pull request by dlipeles**
Submitter Checklist:

- [ ] Submitted a [ticket](https://github.com/brave-intl/publishers/issues) for my issue if one did not already exist.
- [ ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- [ ] Tagged reviewers.

Test Plan:

Reviewer Checklist:

Tests
- [ ] Adequate test coverage exists to prevent regressions

Security:
- [ ] No raw SQL -- Always prefer ActiveRecord query helpers ([more info on StackOverflow](https://stackoverflow.com/questions/41410752/rails-5-sql-injection#41452695))
- [ ] XSS is mitigated -- Avoid `html_safe` and `raw`; escape untrusted content from users and 3rd party APIs ([Rails XSS guide](https://brakemanpro.com/2017/09/08/cross-site-scripting-in-rails) and [OWASP XSS Prevention Cheat Sheet](https://www.owasp.org/index.php/XSS_(Cross_Site_Scripting)_Prevention_Cheat_Sheet))

### CODE REVIEW

File: app/views/publishers/_instant_donation_modal.html.slim [hunk 1]
```diff
@@ -5,6 +5,8 @@
         = image_tag "icn-donation-jar@1x.png", id: 'icn-donation-jar'
         h2.modal-title = t ".headline"
         p = t ".intro"
+        = hidden_field_tag 'preferred_currency', current_publisher.wallet.default_currency
```
Comment (yachtcaptain23) [inline app/views/publishers/_instant_donation_modal.html.slim]: You should check to make sure the default currency is set.

Comment (dlipeles) [reply app/views/publishers/_instant_donation_modal.html.slim]: We may be okay here, it turns out if the user hasn't connected their wallet yet or selected a default currency then the value of `current_publisher.wallet.default_currency` is `USD`.

We need to have some default value for the rewards banner, so I think `USD` would be the most fitting

Comment (dlipeles) [reply app/views/publishers/_instant_donation_modal.html.slim]: Fixed

File: app/views/publishers/_instant_donation_modal.html.slim [hunk 2]
```diff
@@ -5,9 +5,13 @@
...
         p = t ".intro"
-        = hidden_field_tag 'preferred_currency', current_publisher.wallet.default_currency
-        = hidden_field_tag 'conversion_rate', current_publisher.wallet.rates[current_publisher.wallet.default_currency]
         = link_to(t(".continue"), "#", class: 'btn btn-primary', id: "open-banner-button")
+        - if !current_publisher.wallet.default_currency.nil? && !current_publisher.wallet.default_currency.eql?('BAT')
+          = hidden_field_tag 'preferred_currency', current_publisher.wallet.default_currency
+          = hidden_field_tag 'conversion_rate', current_publisher.wallet.rates[current_publisher.wallet.default_currency]
+        - else
+          = hidden_field_tag 'preferred_currency', 'USD'
+          = hidden_field_tag 'conversion_rate', current_publisher.wallet.rates['USD']
```
Comment (yachtcaptain23) [inline app/views/publishers/_instant_donation_modal.html.slim:14]: I think you need `- end`

File: app/views/publishers/_instant_donation_modal.html.slim [hunk 3]
```diff
@@ -6,6 +6,12 @@
         h2.modal-title = t ".headline"
         p = t ".intro"
         = link_to(t(".continue"), "#", class: 'btn btn-primary', id: "open-banner-button")
+        - if !current_publisher.wallet.default_currency.nil? && !current_publisher.wallet.default_currency.eql?('BAT')
```
Comment (yachtcaptain23) [inline app/views/publishers/_instant_donation_modal.html.slim]: `- if current_publisher.wallet.default_currency.present? && !current_publisher.wallet.default_currency.eql?('BAT')`

Comment (dlipeles) [reply app/views/publishers/_instant_donation_modal.html.slim]: Fixed. Should work in Ruby Slim

File: app/views/publishers/_instant_donation_modal.html.slim (hunk 2 above)
Comment (dlipeles) [reply app/views/publishers/_instant_donation_modal.html.slim:14]: Is it the case for Ruby Slim?
[Slim - Control Code](https://www.rubydoc.info/gems/slim/frames#Control_code_-)

Comment (yachtcaptain23) [reply app/views/publishers/_instant_donation_modal.html.slim]: Oh I suppose not.

There's also a crash on https://sentry.io/brave-software/publishers-staging-2/issues/715523160/?query=is:unresolved

Would you make it `- if current_publisher&.wallet&.default_currency.present? && !current_publisher&.wallet&.default_currency.eql?('BAT')`?

**Comment by yachtcaptain23** [comment]
Merged in #23

**Pull request closed without merging**